manifests/v1beta1/components/controller/katib-config.yaml

```diff
@@ -62,6 +62,10 @@ data:
   early-stopping: |-
     {
       "medianstop": {
-        "image": "docker.io/kubeflowkatib/earlystopping-medianstop:latest"
+        "image": "docker.io/kubeflowkatib/earlystopping-medianstop:latest",
+        "algorithmSettings": {
+          "min_trials_required": "int",
+          "start_step": "int"
+        }
```

Comment:
Why did you use Katib config to define these settings ?
For HP or NAS algorithm we use [GRPC API call](https://github.com/kubeflow/katib/blob/master/pkg/apis/manager/v1beta1/api.proto#L36) to validate settings.
Suggestion controller invoke this API when Suggestion deployment is created: https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/suggestion/suggestion_controller.go#L271

I believe for EarlyStopping we can follow the similar approach.
It should be responsibility of Early Stopping service maintainers to define `ValidateEarlyStoppingSettings` _(optional name)_ method to validate settings before `GetEarlyStoppingRules` method runs.

I believe, it will be very hard to be consistent between EarlyStopping Service settings and Katib config.
We might want to keep Suggestion Deployment related setting in Katib config.

What do you think @tenzen-y @gaocegege @johnugeorge ?

Reply:
I defined the parameters to be verified in katibconfig since I think it would be useful if researchers could add new algorithms to the suggestion service and early stopping service without making major changes to the controller.
Also, I think we need to add webhook validation for the values that users may enter.

Reply:
>  researchers and others could add new algorithms to the suggestion service and early stopping service without making major changes to the controller.

That should work with another GRPC API for Early Stopping Algorithm Setting validation, right ?
In that case, researches can easily extend the [Early Stopping Service](https://github.com/kubeflow/katib/blob/master/pkg/earlystopping/v1beta1/medianstop/service.py) without changing the Controller source code.

Reply:
Similar to Suggestion service, for example [SKOpt Suggestion service](https://github.com/kubeflow/katib/blob/master/pkg/suggestion/v1beta1/skopt/service.py#L60-L67).